### PR TITLE
Web Frontend

### PR DESCRIPTION
**Pull request by nateyazdani**
First off, let me apologize for the sheer size of this pull request. This has been brewing for quite a long time, and I had originally expected it to be a lot more lightweight. I promise that the majority of the code is either super simple or from a reputable third-party (_e.g._, jQuery, Bootstrap). Aside from that, this is pretty much ~~ripping off~~ inspired by datagrepper's website. To give you an idea of what this looks like, the home page is pictured below. (Note: I found a bug halfway through writing this, but I haven't been able to send a PR due to having such a, ahem, polluted working tree. That PR is now pending.)

image

### CODE REVIEW

**Comment by ralphbean** [comment]
That frontpage looks awesome!  :p

File: setup.py [hunk 1]
```diff
@@ -20,6 +25,20 @@ def get_description():
     'freezegun'
 ]
 
+# TODO: let a flag or option determine whether to actually install these
+install_data = [
```
Comment (ralphbean) [inline setup.py]: Yeah, watch out for this.  This makes things _really_ screwy when you try to pip install the project into a virtualenv.  We've tried doing it in a handful of other projects and always ended up backing it out and relying on .spec files to handle `/usr/share` and `/etc`.

File: statscache/templates/reference.html [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+{% block body %}
+    <section class="lead">
+        <h1>overview</h1>
+        <p>
+            To allow integration with other web applications and services,
+            statscache offers a simple, easy-to-use REST API with support for
+            multiple convenient data formats.
+        </p>
+    </section>
+{# This page really ought to be more thorough #}
```
Comment (ralphbean) [inline statscache/templates/reference.html:23]: It's a good start.  We can revise it as we learn what people get confused about (via what questions they come and ask us).

**Comment by ralphbean** [comment]
:+1:, but I'd consider removing that `install_data` bit in setup.py.. that's my only beef.  It looks convenient, but it ends up the opposite.

File: statscache/templates/reference.html (hunk 2 above)
Comment (nateyazdani) [inline statscache/templates/reference.html:23]: Sounds good :) part of me really wanted to keep at it, but I'd already spent days on just that page (not much of a writer)

**Comment by nateyazdani** [comment]
Thanks for the feedback :) And that's a bummer about `install_data`. I was curious about how it might interact with virtualenvs, and I guess the answer is not too well :/ I'll take that out and add a note in `setup.py` to remind packagers to do soemthing like that.

**Comment by ralphbean** [comment]
:+1: to merge!

**Pull request merged**